### PR TITLE
[Components with more than 1 service] LPS-181736 Refactor PortletToolbarContributor components from Echo modules

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://issues.liferay.com/browse/LPS-181736

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:echo

**Comment by catarineaguiar** [comment]
@MrJohn1911 semver error:

```
     [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/document-library/document-library-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.document.library.portlet.toolbar.contributor MAJOR      1.2.0      1.2.0      2.0.0      VERSION INCREASE REQUIRED
     [exec]     	-   interface  com.liferay.document.library.portlet.toolbar.contributor.DLPortletToolbarContributor
     [exec]     		-   access     abstract
     [exec]     		-   implements com.liferay.portal.kernel.portlet.toolbar.contributor.PortletToolbarContributor
     [exec]     		-   method     clone()
     [exec]     			-   access     protected
     [exec]     			-   return     java.lang.Object
     [exec]     		-   method     equals(java.lang.Object)
     [exec]     			-   return     boolean
     [exec]     		-   method     finalize()
     [exec]     			-   access     protected
     [exec]     			-   return     void
     [exec]     		-   method     getPortletTitleAddDocumentMenuItems(com.liferay.portal.kernel.repository.model.Folder,com.liferay.portal.kernel.theme.ThemeDisplay,javax.portlet.PortletRequest)
     [exec]     			-   return     java.util.List
     [exec]     		-   method     getPortletTitleAddFolderMenuItem(com.liferay.portal.kernel.theme.ThemeDisplay,javax.portlet.PortletRequest,com.liferay.portal.kernel.repository.model.Folder)
     [exec]     			-   return     com.liferay.portal.kernel.servlet.taglib.ui.MenuItem
     [exec]     		-   method     getPortletTitleAddMultipleDocumentsMenuItem(com.liferay.portal.kernel.theme.ThemeDisplay,javax.portlet.PortletRequest,com.liferay.portal.kernel.repository.model.Folder)
     [exec]     			-   return     com.liferay.portal.kernel.servlet.taglib.ui.MenuItem
     [exec]     		-   method     hashCode()
     [exec]     			-   return     int
     [exec]     		-   method     toString()
     [exec]     			-   return     java.lang.String
     [exec]     [Baseline Warning] Bundle Version Change Recommended: 20.0.0
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-37/test-portal-acceptance-pullrequest(master)/11181/modules-semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
@MrJohn1911 The semver issue is fixed. No regressions.

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/toolbar/contributor/DefaultDLPortletToolbarContributor.java [hunk 1]
```diff
@@ -54,13 +53,10 @@
...
 	},
-	service = {
-		DLPortletToolbarContributor.class, PortletToolbarContributor.class
-	}
+	service = PortletToolbarContributor.class
 )
 public class DefaultDLPortletToolbarContributor
-	extends BasePortletToolbarContributor
-	implements DLPortletToolbarContributor {
+	extends BasePortletToolbarContributor {
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/toolbar/contributor/DefaultDLPortletToolbarContributor.java:59]: Hey @MrJohn1911, rename it to DLPortletToolbarContributor. Thx

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/SearchMVCRenderCommand.java [hunk 2]
```diff
@@ -64,20 +64,20 @@ public String render(
...
 		renderRequest.setAttribute(
 			DLWebKeys.DOCUMENT_LIBRARY_TRASH_HELPER, _dlTrashHelper);
 
 		return "/document_library/view.jsp";
 	}
 
-	@Reference
-	private DLAdminDisplayContextProvider _dlAdminDisplayContextProvider;
+	@Reference(
+		target = "(component.name=com.liferay.document.library.web.internal.portlet.toolbar.contributor.DefaultDLPortletToolbarContributor)"
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/SearchMVCRenderCommand.java:75]: Hey @MrJohn1911, filter using the `javax.portlet.name` property instead. Thx

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/DLViewMVCRenderCommand.java [hunk 3]
```diff
@@ -186,6 +185,11 @@ private boolean _pingFolderRepository(
 	@Reference
 	private DDMStructureService _ddmStructureService;
 
+	@Reference(
+		target = "(component.name=com.liferay.document.library.web.internal.portlet.toolbar.contributor.DefaultDLPortletToolbarContributor)"
+	)
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/DLViewMVCRenderCommand.java:190]: Hey @MrJohn1911, filter using the `javax.portlet.name` property instead. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:echo

**Comment by zhangyan0701** [comment]
Hi @MrJohn1911 Please resolve the conflict and rerun the tests, thanks!

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:echo

**Comment by catarineaguiar** [comment]
@MrJohn1911 regression found:

```
    [junit] Testcase: testGetPortletRenderPartsWithFastLoad(com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest):	FAILED
    [junit] Actual string /combo?minifierType=css&themeId=theme_id&com.liferay.portlet.1:/o/portlet-web/header-portlet.css&/header-portal.css&t=8
    [junit] junit.framework.AssertionFailedError: Actual string /combo?minifierType=css&themeId=theme_id&com.liferay.portlet.1:/o/portlet-web/header-portlet.css&/header-portal.css&t=8
    [junit] 	at com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest._assertEquals(PortletRenderUtilTest.java:279)
    [junit] 	at com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest.testGetPortletRenderPartsWithFastLoad(PortletRenderUtilTest.java:224)
    [junit] 
    [junit] 
    [junit] Testcase: testGetPortletRenderPartsWithContextAndFastLoad(com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest):	FAILED
    [junit] Actual string /portal/combo?minifierType=css&themeId=theme_id&com.liferay.portlet.1:/portal/o/portlet-web/header-portlet.css&/portal/header-portal.css&t=8
    [junit] junit.framework.AssertionFailedError: Actual string /portal/combo?minifierType=css&themeId=theme_id&com.liferay.portlet.1:/portal/o/portlet-web/header-portlet.css&/portal/header-portal.css&t=8
    [junit] 	at com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest._assertEquals(PortletRenderUtilTest.java:279)
    [junit] 	at com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest.testGetPortletRenderPartsWithContextAndFastLoad(PortletRenderUtilTest.java:165)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-36/test-portal-acceptance-pullrequest(master)/3591/unit-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

semver error:
```
     [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/document-library/document-library-api] FAILED
     [exec]     java.lang.AssertionError: [Baseline Warning] Bundle Version Change Recommended: 21.0.0
     [exec]         at org.junit.Assert.fail(Assert.java:88)
     [exec]         at org.junit.Assert.assertTrue(Assert.java:41)
     [exec]         at org.junit.Assert.assertFalse(Assert.java:64)
     [exec]         at com.liferay.semantic.versioning.SemanticVersioningTest.testSemanticVersioning(SemanticVersioningTest.java:82)
     [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-36/test-portal-acceptance-pullrequest(master)/3591/modules-semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
> @MrJohn1911 regression found:
> 
> ```
>     [junit] Testcase: testGetPortletRenderPartsWithFastLoad(com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest):	FAILED
>     [junit] Actual string /combo?minifierType=css&themeId=theme_id&com.liferay.portlet.1:/o/portlet-web/header-portlet.css&/header-portal.css&t=8
>     [junit] junit.framework.AssertionFailedError: Actual string /combo?minifierType=css&themeId=theme_id&com.liferay.portlet.1:/o/portlet-web/header-portlet.css&/header-portal.css&t=8
>     [junit] 	at com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest._assertEquals(PortletRenderUtilTest.java:279)
>     [junit] 	at com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest.testGetPortletRenderPartsWithFastLoad(PortletRenderUtilTest.java:224)
>     [junit] 
>     [junit] 
>     [junit] Testcase: testGetPortletRenderPartsWithContextAndFastLoad(com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest):	FAILED
>     [junit] Actual string /portal/combo?minifierType=css&themeId=theme_id&com.liferay.portlet.1:/portal/o/portlet-web/header-portlet.css&/portal/header-portal.css&t=8
>     [junit] junit.framework.AssertionFailedError: Actual string /portal/combo?minifierType=css&themeId=theme_id&com.liferay.portlet.1:/portal/o/portlet-web/header-portlet.css&/portal/header-portal.css&t=8
>     [junit] 	at com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest._assertEquals(PortletRenderUtilTest.java:279)
>     [junit] 	at com.liferay.portal.kernel.portlet.render.PortletRenderUtilTest.testGetPortletRenderPartsWithContextAndFastLoad(PortletRenderUtilTest.java:165)
> ```
> 
> full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-36/test-portal-acceptance-pullrequest(master)/3591/unit-jdk8/0/0/jenkins-console.txt.gz?authuser=0)
> 
> semver error:
> 
> ```
>      [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/document-library/document-library-api] FAILED
>      [exec]     java.lang.AssertionError: [Baseline Warning] Bundle Version Change Recommended: 21.0.0
>      [exec]         at org.junit.Assert.fail(Assert.java:88)
>      [exec]         at org.junit.Assert.assertTrue(Assert.java:41)
>      [exec]         at org.junit.Assert.assertFalse(Assert.java:64)
>      [exec]         at com.liferay.semantic.versioning.SemanticVersioningTest.testSemanticVersioning(SemanticVersioningTest.java:82)
>      [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>      [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> ```
> 
> full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-36/test-portal-acceptance-pullrequest(master)/3591/modules-semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

Hey @catarineaguiar, I think this regression is also occurring in master, can you please double check it? thanks in advance!

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Review by rafaprax — APPROVED** [review]

**Comment by rafaprax** [comment]
Hey @MrJohn1911, next time try applying, when possible, the renaming changes along with the others on the same files.
Thx

**Comment by MrJohn1911** [comment]
Last push was just a baseline conflict.

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
ci:stop:sf

**Comment by catarineaguiar** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
no regressions found, just needing sf results

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
ci:forward:force

cc @shuyangzhou @tinatian

**Pull request closed without merging**